kubernetes-helm fails to build

**Describe the bug**
When I attempt to use `kubernetes-helm` from `nix run` or `nix-shell`, I get the following error:

```
$ nix run nixpkgs.kubernetes-helm
builder for '/nix/store/yrizzpzasyr2rdnzx0hgcyrd4bgsbrcx-helm-3.2.1.drv' failed with exit code 1; last 10 log lines:
  pkg/helmpath/lazypath_unix.go:21:2: cannot find package "." in:
         /build/source/vendor/k8s.io/client-go/util/homedir
  cmd/helm/helm.go:29:2: cannot find package "." in:
        /build/source/vendor/k8s.io/klog
  pkg/kube/client.go:48:2: cannot find package "." in:
        /build/source/vendor/k8s.io/kubectl/pkg/cmd/util
  pkg/kube/factory.go:23:2: cannot find package "." in:
       /build/source/vendor/k8s.io/kubectl/pkg/validation
  pkg/chart/loader/load.go:27:2: cannot find package "." in:
        /build/source/vendor/sigs.k8s.io/yaml
[0 built (1 failed)]
error: build of '/nix/store/yrizzpzasyr2rdnzx0hgcyrd4bgsbrcx-helm-3.2.1.drv' failed
```

**To Reproduce**
Steps to reproduce the behavior:
1. `nix run nixpkgs.kubernetes-helm`

**Expected behavior**
Either `helm` is downloaded from cache or builds successfully.

**Additional context**
I had originally thought this was due to the recent [GitHub outage](https://www.githubstatus.com/incidents/shdhxntw519h), but it appears that there may be some upstream issues with Helm. For example, the [release notes for v3.2.2](https://github.com/helm/helm/releases/tag/v3.2.2) say this:

> This release is marked as a security release due to a nuance with the Go module proxy. During the last release, a Git tag was accidentally created, then deleted, and then re-created correctly. In that time, the Go proxy cached the wrong version. Even now, a month later, the Google-run proxy has an invalid SHA. Thus, when a build from source occurs, in some rare cases the user may get a security error directly from the Go toolchain.

That having been said, I'm not totally convinced this is the root cause. I tried to confirm the fix with 3.2.2 by overriding the helm derivation to use the new version:

```nix
# To build:
# nix-build --argstr helmVersion 3.2.2
{
  pkgs ? import <nixpkgs> {}
, helmVersion
}:
let
  inherit (pkgs) fetchFromGitHub kubernetes-helm lib;

  hashes = {
    "3.2.2" = "1jy3xffczhf1ddpj0vzmp49l7mhqj7pjz5prsywg1jdk8hc6dyw1";
    "3.2.3" = "1ds682l8y6aaz57jvpj7kn4ja1hvxc1dl6804ymsqwjqx8rs33ry";
    "3.2.4" = "1plpk8qnv11d47qz93h57abjchyp6ahgyazyp0c6rv24vb9fp9zi";
  };

  version = "3.2.4";

  src = fetchFromGitHub {
    owner = "helm";
    repo = "helm";
    rev = "v${version}";
    sha256 = hashes.${version};
  };

in
kubernetes-helm.overrideDerivation (_: {
  name = "helm-${version}";
  inherit version;
  inherit src;
})
```

No matter which of 3.2.2, 3.2.3, or 3.2.4 I choose, I get the same set of errors (see collapsed element below).

<details>
<summary><strong>Build log for 3.2.4</strong></summary>

```
these derivations will be built:
  /nix/store/gpkk5qr1pakkypfgb6lg17b0xl6slmra-helm-3.2.4.drv
building '/nix/store/gpkk5qr1pakkypfgb6lg17b0xl6slmra-helm-3.2.4.drv'...
unpacking sources
unpacking source archive /nix/store/ysvwnjjpyy915xrr3d5x1hkv78j6viwv-source
source root is source
patching sources
configuring
building
Building subPackage ./cmd/helm
pkg/engine/funcs.go:25:2: cannot find package "." in:
        /build/source/vendor/github.com/BurntSushi/toml
cmd/helm/search_repo.go:29:2: cannot find package "." in:
        /build/source/vendor/github.com/Masterminds/semver/v3
pkg/action/install.go:30:2: cannot find package "." in:
        /build/source/vendor/github.com/Masterminds/sprig/v3
pkg/storage/driver/sql.go:27:2: cannot find package "." in:
        /build/source/vendor/github.com/Masterminds/squirrel
pkg/plugin/installer/vcs_installer.go:23:2: cannot find package "." in:
        /build/source/vendor/github.com/Masterminds/vcs
pkg/lint/rules/chartfile.go:25:2: cannot find package "." in:
        /build/source/vendor/github.com/asaskevich/govalidator
internal/experimental/registry/cache.go:29:2: cannot find package "." in:
        /build/source/vendor/github.com/containerd/containerd/content
internal/experimental/registry/cache.go:30:2: cannot find package "." in:
        /build/source/vendor/github.com/containerd/containerd/errdefs
internal/experimental/registry/resolver.go:20:2: cannot find package "." in:
        /build/source/vendor/github.com/containerd/containerd/remotes
pkg/chartutil/expand.go:25:2: cannot find package "." in:
        /build/source/vendor/github.com/cyphar/filepath-securejoin
internal/experimental/registry/authorizer.go:20:2: cannot find package "." in:
        /build/source/vendor/github.com/deislabs/oras/pkg/auth
internal/experimental/registry/client.go:27:2: cannot find package "." in:
        /build/source/vendor/github.com/deislabs/oras/pkg/auth/docker
internal/experimental/registry/cache.go:31:2: cannot find package "." in:
        /build/source/vendor/github.com/deislabs/oras/pkg/content
internal/experimental/registry/util.go:25:2: cannot find package "." in:
        /build/source/vendor/github.com/deislabs/oras/pkg/context
internal/experimental/registry/client.go:28:2: cannot find package "." in:
        /build/source/vendor/github.com/deislabs/oras/pkg/oras
cmd/helm/registry_login.go:28:2: cannot find package "." in:
        /build/source/vendor/github.com/docker/docker/pkg/term
internal/experimental/registry/util.go:26:2: cannot find package "." in:
        /build/source/vendor/github.com/docker/go-units
pkg/kube/client.go:28:2: cannot find package "." in:
        /build/source/vendor/github.com/evanphx/json-patch
pkg/engine/files.go:24:2: cannot find package "." in:
        /build/source/vendor/github.com/gobwas/glob
cmd/helm/repo_add.go:29:2: cannot find package "." in:
        /build/source/vendor/github.com/gofrs/flock
cmd/helm/history.go:25:2: cannot find package "." in:
        /build/source/vendor/github.com/gosuri/uitable
pkg/storage/driver/sql.go:24:2: cannot find package "." in:
        /build/source/vendor/github.com/jmoiron/sqlx
pkg/storage/driver/sql.go:30:2: cannot find package "." in:
        /build/source/vendor/github.com/lib/pq
pkg/chartutil/coalesce.go:22:2: cannot find package "." in:
        /build/source/vendor/github.com/mitchellh/copystructure
internal/experimental/registry/cache.go:32:2: cannot find package "." in:
        /build/source/vendor/github.com/opencontainers/go-digest
internal/experimental/registry/cache.go:33:2: cannot find package "." in:
        /build/source/vendor/github.com/opencontainers/image-spec/specs-go
internal/experimental/registry/cache.go:34:2: cannot find package "." in:
        /build/source/vendor/github.com/opencontainers/image-spec/specs-go/v1
cmd/helm/completion.go:24:2: cannot find package "." in:
        /build/source/vendor/github.com/pkg/errors
pkg/storage/driver/sql.go:25:2: cannot find package "." in:
        /build/source/vendor/github.com/rubenv/sql-migrate
internal/experimental/registry/util.go:27:2: cannot find package "." in:
        /build/source/vendor/github.com/sirupsen/logrus
cmd/helm/chart.go:21:2: cannot find package "." in:
        /build/source/vendor/github.com/spf13/cobra
cmd/helm/docs.go:24:2: cannot find package "." in:
        /build/source/vendor/github.com/spf13/cobra/doc
cmd/helm/flags.go:24:2: cannot find package "." in:
        /build/source/vendor/github.com/spf13/pflag
pkg/chartutil/jsonschema.go:25:2: cannot find package "." in:
        /build/source/vendor/github.com/xeipuuv/gojsonschema
pkg/provenance/sign.go:29:2: cannot find package "." in:
        /build/source/vendor/golang.org/x/crypto/openpgp
pkg/provenance/sign.go:30:2: cannot find package "." in:
        /build/source/vendor/golang.org/x/crypto/openpgp/clearsign
pkg/provenance/sign.go:31:2: cannot find package "." in:
        /build/source/vendor/golang.org/x/crypto/openpgp/packet
pkg/action/package.go:27:2: cannot find package "." in:
        /build/source/vendor/golang.org/x/crypto/ssh/terminal
internal/third_party/k8s.io/kubernetes/deployment/util/deploymentutil.go:23:2: cannot find package "." in:
        /build/source/vendor/k8s.io/api/apps/v1
pkg/kube/wait.go:26:2: cannot find package "." in:
        /build/source/vendor/k8s.io/api/apps/v1beta1
pkg/kube/wait.go:27:2: cannot find package "." in:
        /build/source/vendor/k8s.io/api/apps/v1beta2
pkg/kube/client.go:30:2: cannot find package "." in:
        /build/source/vendor/k8s.io/api/batch/v1
internal/third_party/k8s.io/kubernetes/deployment/util/deploymentutil.go:24:2: cannot find package "." in:
        /build/source/vendor/k8s.io/api/core/v1
pkg/kube/wait.go:30:2: cannot find package "." in:
        /build/source/vendor/k8s.io/api/extensions/v1beta1
pkg/kube/client.go:32:2: cannot find package "." in:
        /build/source/vendor/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1
pkg/kube/client.go:33:2: cannot find package "." in:
        /build/source/vendor/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1
internal/third_party/k8s.io/kubernetes/deployment/util/deploymentutil.go:25:2: cannot find package "." in:
        /build/source/vendor/k8s.io/apimachinery/pkg/api/equality
pkg/kube/client.go:34:2: cannot find package "." in:
        /build/source/vendor/k8s.io/apimachinery/pkg/api/errors
pkg/kube/client.go:36:2: cannot find package "." in:
        /build/source/vendor/k8s.io/apimachinery/pkg/api/meta
internal/third_party/k8s.io/kubernetes/deployment/util/deploymentutil.go:26:2: cannot find package "." in:
        /build/source/vendor/k8s.io/apimachinery/pkg/apis/meta/v1
pkg/kube/client.go:38:2: cannot find package "." in:
        /build/source/vendor/k8s.io/apimachinery/pkg/fields
pkg/kube/wait.go:34:2: cannot find package "." in:
        /build/source/vendor/k8s.io/apimachinery/pkg/labels
pkg/kube/client.go:39:2: cannot find package "." in:
        /build/source/vendor/k8s.io/apimachinery/pkg/runtime
pkg/kube/converter.go:26:2: cannot find package "." in:
        /build/source/vendor/k8s.io/apimachinery/pkg/runtime/schema
pkg/kube/client.go:40:2: cannot find package "." in:
        /build/source/vendor/k8s.io/apimachinery/pkg/types
internal/third_party/k8s.io/kubernetes/deployment/util/deploymentutil.go:27:2: cannot find package "." in:
        /build/source/vendor/k8s.io/apimachinery/pkg/util/intstr
pkg/kube/client.go:41:2: cannot find package "." in:
        /build/source/vendor/k8s.io/apimachinery/pkg/util/strategicpatch
pkg/storage/driver/cfgmaps.go:30:2: cannot find package "." in:
        /build/source/vendor/k8s.io/apimachinery/pkg/util/validation
pkg/kube/wait.go:37:2: cannot find package "." in:
        /build/source/vendor/k8s.io/apimachinery/pkg/util/wait
pkg/kube/client.go:42:2: cannot find package "." in:
        /build/source/vendor/k8s.io/apimachinery/pkg/watch
pkg/kube/client.go:43:2: cannot find package "." in:
        /build/source/vendor/k8s.io/cli-runtime/pkg/genericclioptions
pkg/kube/client.go:44:2: cannot find package "." in:
        /build/source/vendor/k8s.io/cli-runtime/pkg/resource
pkg/engine/lookup_func.go:28:2: cannot find package "." in:
        /build/source/vendor/k8s.io/client-go/discovery
pkg/engine/lookup_func.go:29:2: cannot find package "." in:
        /build/source/vendor/k8s.io/client-go/dynamic
pkg/kube/factory.go:21:2: cannot find package "." in:
        /build/source/vendor/k8s.io/client-go/kubernetes
pkg/kube/client.go:45:2: cannot find package "." in:
        /build/source/vendor/k8s.io/client-go/kubernetes/scheme
internal/third_party/k8s.io/kubernetes/deployment/util/deploymentutil.go:28:2: cannot find package "." in:
        /build/source/vendor/k8s.io/client-go/kubernetes/typed/apps/v1
pkg/storage/driver/cfgmaps.go:31:2: cannot find package "." in:
        /build/source/vendor/k8s.io/client-go/kubernetes/typed/core/v1
cmd/helm/helm.go:33:2: cannot find package "." in:
        /build/source/vendor/k8s.io/client-go/plugin/pkg/client/auth
pkg/engine/engine.go:30:2: cannot find package "." in:
        /build/source/vendor/k8s.io/client-go/rest
pkg/kube/client.go:46:2: cannot find package "." in:
        /build/source/vendor/k8s.io/client-go/tools/cache
pkg/kube/factory.go:22:2: cannot find package "." in:
        /build/source/vendor/k8s.io/client-go/tools/clientcmd
pkg/kube/client.go:47:2: cannot find package "." in:
        /build/source/vendor/k8s.io/client-go/tools/watch
pkg/helmpath/lazypath_unix.go:21:2: cannot find package "." in:
        /build/source/vendor/k8s.io/client-go/util/homedir
cmd/helm/helm.go:29:2: cannot find package "." in:
        /build/source/vendor/k8s.io/klog
pkg/kube/client.go:48:2: cannot find package "." in:
        /build/source/vendor/k8s.io/kubectl/pkg/cmd/util
pkg/kube/factory.go:23:2: cannot find package "." in:
        /build/source/vendor/k8s.io/kubectl/pkg/validation
pkg/chart/loader/load.go:27:2: cannot find package "." in:
        /build/source/vendor/sigs.k8s.io/yaml
builder for '/nix/store/gpkk5qr1pakkypfgb6lg17b0xl6slmra-helm-3.2.4.drv' failed with exit code 1
error: build of '/nix/store/gpkk5qr1pakkypfgb6lg17b0xl6slmra-helm-3.2.4.drv' failed
```

</details>

**Notify maintainers**

```
> map (m: m.github) kubernetes-helm.meta.maintainers
[ "rlupton20" "edude03" "saschagrunert" "Frostman" "kalbasit" ]
```

- @rlupton20
- @edude03
- @saschagrunert
- @Frostman
- @kalbasit

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.0-7634-generic, Pop!_OS, 20.04 LTS`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.4`
 - channels(gustavderdrache): `"nixpkgs-20.09pre230446.a84cbb60f02"`
 - nixpkgs: `/home/gustavderdrache/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/applications/networking/cluster/helm/default.nix
+++ b/pkgs/applications/networking/cluster/helm/default.nix
@@ -10,7 +10,7 @@ buildGoModule rec {
     rev = "v${version}";
     sha256 = "1453qkd9s4z4r0xzmv8ym7qfg33szf6gizfkb5zxj590fcbsgnd7";
   };
-  vendorSha256 = null;
+  vendorSha256 = "0j25m56cwzjd9b75v7xlb26q81bsmln77k23h9n8v2f2gqwwpkrl";
 
   subPackages = [ "cmd/helm" ];
   buildFlagsArray = [ "-ldflags=-w -s -X helm.sh/helm/v3/internal/version.version=v${version}" ];

